main.go

```diff
@@ -222,13 +223,30 @@ func main() {
 	result, platformsDetected := scanner.GenerateScanResult(searchDir, hasSSHKey)
 
 	// Upload results
-	if resultClient != nil {
-		log.TInfof("Submitting results...")
-		if err := resultClient.uploadResults(result); err != nil {
-			failf("Could not send back results: %s", err)
+	var resultBytes []byte
+	shouldStoreResult := resultClient != nil || strings.HasPrefix(cfg.ResultSubmitURL, "path::")
```

Comment:
I like how you extracted this into a named boolean variable which helps readability a lot. Can you do the same with the path prefix check because it is also used at 4 different places and if we name it then it will be much faster to understand what it means. Something like
```
isLocalSave := strings.HasPrefix(cfg.ResultSubmitURL, "path::")
```

Reply:
Updated in https://github.com/bitrise-steplib/steps-project-scanner/pull/148/commits/bdcb0dae4a42cf642571062fc889b7db71efd8b4